docs/reference/TensorFlow.md

```diff
@@ -1,405 +1,394 @@
-```
-You need to add the include from the following snippet and change the variable declaration without changing the hex values:
-```
-#include "tensorflow/lite/micro/examples/micro_speech/micro_features/model.h"  
-const unsigned char g_tiny_conv_micro_features_model_data[] = {  
-0x18, 0x00, 0x00, 0x00, 0x54, 0x46, 0x4c, 0x33, 0x00, 0x00, 0x0e, 0x00,
-```
-Next, go to the very bottom of the file and find the unsigned int variable.
-```
-unsigned int tiny_conv_tflite_len = 18216;
-```
-Change the declaration to the following code, but do not change the number assigned to it, even if your number is different from the one in this guide.
-```
-const int g_tiny_conv_micro_features_model_data_len = 18216;
-```
-Finally, save the file, then copy the ```tiny_conv_micro_features_model_data.cc```file into the ```tensorflow/tensorflow/lite/micro/tools/make/gen/mbed_cortex-m4/prj/micro_speech/mbed/tensorflow/lite/micro/examples/micro_speech/micro_features``` directory.
+    ```
+    void RespondToCommand(tflite::ErrorReporter* error_reporter,  
+    int32_t current_time, const char* found_command,  
+    uint8_t score, bool is_new_command) {  
+        if (is_new_command) {  
+            error_reporter->Report("Heard %s (%d) @%dms", found_command, score,  
+            current_time);  
+        if(strcmp(found_command, "up") == 0) {  
+            lcd. Clear(0xFF0F9D58);  
+            lcd.DisplayStringAt(0, LINE(5), (uint8_t *)"Heard up", CENTER_MODE);  
+        } else if(strcmp(found_command, "down") == 0) {  
+            lcd.Clear(0xFFDB4437);  
+            lcd.DisplayStringAt(0, LINE(5), (uint8_t *)"Heard down", CENTER_MODE);  
+        } else if(strcmp(found_command, "unknown") == 0) {  
+            lcd.Clear(0xFFF4B400);  
+            lcd.DisplayStringAt(0, LINE(5), (uint8_t *)"Heard unknown", CENTER_MODE);  
+        } else {  
+            lcd.Clear(0xFF4285F4);  
+            lcd.DisplayStringAt(0, LINE(5), (uint8_t *)"Heard silence", CENTER_MODE);  
+            }  
+        }  
+    }
+    ```
+<!--I added indents - I assumed their lack was a result of copy/paste destroying something, not a decision-->
 
-# Modify the device code
+1. Navigate back to the mbed directory:
 
-If you build and run your code now, your device should respond to the words “up” and “down”. However, the code was written to assume that the words are “yes” and “no”. Let’s update the references and the user interface so that the appropriate words are printed.
+    ```
+    cd <path_to_tensorflow>/tensorflow/lite/micro/tools/make/gen/mbed_cortex-m4/prj/micro_speech/mbed
+    ```
 
-First, go to the following directory:
+1. Rebuilt the project:
 
-```tensorflow/lite/micro/examples/micro_speech/```
+    ```
+    mbed compile -m K66F -t GCC_ARM
+    ```
 
-and open the file:
+1. Copy the binary to the USB storage of the device<!--why aren't we using flash?-->, using the same method that you used earlier.
```

Comment:
I guess we can leave this out and show it as before. I'm not sure what the logic behind showing two ways of doing it was when the original guide was made.

Reply:
```suggestion

```